stdlib/Dates/src/types.jl

```diff
@@ -310,6 +310,26 @@ function Time(period::TimePeriod, periods::TimePeriod...)
     return Time(h, mi, s, ms, us, ns)
 end
 
+# Convenience constructor for DateTime from Date and Time
+"""
+    DateTime(d::Date, t::Time)
+
+Construct a `DateTime` type by `Date` and `Time`.
+
+```jldoctest
+julia> d = Date(2018, 1, 1)
+2018-01-01
+
+julia> t = Time(8, 15, 42)
+08:15:42
+
+julia> DateTime(d, t)
+2018-01-01T08:15:42
+```
+"""
+DateTime(d::Date, t::Time) =
+    DateTime(UTM(value(d) * 86400000 + value(t) / 1000000))
```

Comment:
Hmm, this isn't quite right because `Time` internal values aren't normalized down; i.e. `Dates.value(Time(0,0,0)) != Dates.value(Time(0,0,0) + Dates.Hour(24))`. I would maybe just mirror the definition of `+(d::Date, t::Time)` [here](https://github.com/JuliaLang/julia/blob/70be5f25981918af138cce21bace331aa2d16945/stdlib/Dates/src/arithmetic.jl#L21)

Reply:
If I define it as `DateTime(UTM(value(d) * 86400000 + (value(t) / 1000000) % 86400000))`,
seems works.
There are no significant performance differences between mine and `+`, so I mirrored the `+(d::Date, t::Time)`.

Reply:
It seems a bit backwards to me to have something in `types.jl` depend on something `arithmetic.jl`; would you mind switching the definitions around? i.e. have `types.jl` define the actual definition and then change the `arithmetic.jl` definition to just call `DateTime(dt, t)`? It might just save some headache down the road if further refactoring is done (keeping the dependency graph clean and all that...)